File: 移动零.cpp
/*
* 双指针， 时间复杂度O(n) 空间复杂度O(1)
* 
*/
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int lastNoneZeroPoint = 0;
        for(int i = 0; i < nums.size(); i++)
        {
            if(nums[i] != 0)
            {
                nums[lastNoneZeroPoint++] = nums[i];
            }
        }
        
        for(int i = lastNoneZeroPoint; i < nums.size(); i++)
        {
            nums[i] = 0;
        }
    }
};

File: 移动零.py
"""
思路挺简单的，就是我们先遍历一边nums，统计0的个数，在遍历的过程中删除每一个找到的0，最后只需要在最后添加多少个零就好
"""
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0 
        count = 0
        while i < len(nums):
            if nums[i] == 0:
                count += 1
                del nums[i]
            else:
                i += 1
                
        for i in range(count):
            nums.append(0)
                
